### PR TITLE
Fix infinite ticker loop

### PR DESCRIPTION
**Pull request by tacaswell**
This just prevents the infinite loop.

The code this touches dates from 2006 so this possible infinite loop has been there for a long time.

### CODE REVIEW

**Comment by WeatherGod** [comment]
The pdf version of this test failed: FAIL: matplotlib.tests.test_axes.test_markevery_log_scales.test

**Comment by tacaswell** [comment]
That is very odd....

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -1575,6 +1578,9 @@ def tick_values(self, vmin, vmax):
             subs = self._subs
 
         stride = 1
+        if not self.numticks > 1:
+            raise RuntimeError('The number of ticks must be greater than 1 '
+                               'for LogLocator.')
         while numdec / stride + 1 > self.numticks:
```
Comment (WeatherGod) [inline lib/matplotlib/ticker.py]: Is numdec calculated correctly up above? It floors vmax, and ceils vmin, but down below, it is the other way around.

Comment (tacaswell) [inline lib/matplotlib/ticker.py]: I spent a while trying to understand this code and gave up.  The blame on it is all pretty old (2006-2008)  so I took a pass on touching it and was trying to just prevent the infinite loop.

**Comment by dopplershift** [comment]
The virtue of the code itself is certainly questionable, but this change seems fine by itself.

**Pull request merged**